### PR TITLE
Fix and simplify examples

### PR DESCRIPTION
**Pull request by joaquimg**
I think we should only keep the second experiment.

### CODE REVIEW

File: docs/Project.toml [hunk 1]
```diff
@@ -9,3 +9,4 @@ OSQP = "ab2f91bb-94b4-55e3-9ba0-7f65df51de79"
 Plots = "91a5bcdd-55d7-5caf-9e0b-520d859cae80"
 SCS = "c946c3f1-0d1f-5ce8-9dea-7daa1f7e2d13"
 Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+Ipopt = "b6b21f68-93f8-5de0-b562-5493be1d77c9"
```
Comment (matbesancon) [inline docs/Project.toml:12]:
```suggestion
Ipopt = "b6b21f68-93f8-5de0-b562-5493be1d77c9"
```
```suggestion
Ipopt = "b6b21f68-93f8-5de0-b562-5493be1d77c9"

```

**Comment by matbesancon** [comment]
thanks @joaquimg, the second experiment out of which and which?

File: docs/src/examples/sensitivity-analysis-svm.jl [hunk 2]
```diff
@@ -94,59 +98,27 @@ Plots.plot!(p, svm_x, svm_y, label = "loss = $(round(loss, digits=2))", width=3)
...
-    )
+    for j in 1:N
+        MOI.set(
+            model,
+            DiffOpt.ForwardInConstraint(),
+            cons[j],
+            dy[j] * (dot(X[j,:], index.(w)) + index(b)),
+        )
+        # analogous to varying theta in the expression:
+        # (y_{i} + theta) (w^T X_{i} + b) \ge 1 - \xi_{i}
```
Comment (matbesancon) [inline docs/src/examples/sensitivity-analysis-svm.jl:120]: there is a bug in Literate it seems where having a comment inside a loop will just split the code block and add text in the middle
```suggestion
```

File: docs/src/examples/sensitivity-analysis-svm.jl [hunk 3]
```diff
@@ -94,59 +98,27 @@ Plots.plot!(p, svm_x, svm_y, label = "loss = $(round(loss, digits=2))", width=3)
...
 # ## Experiment 1: Gradient of hyperplane wrt the data point labels
 
 # Construct perturbations in data point labels `y` without changing the data point coordinates `X`.
 
-
-∇ = Float64[]
+∇ = zeros(N)
 dy = zeros(N);
 
 # begin differentiating
```
Comment (matbesancon) [inline docs/src/examples/sensitivity-analysis-svm.jl:108]:
```suggestion
# Begin differentiating the model.
# analogous to varying theta in the expression:
# ```math
# (y_{i} + \theta) (w^T X_{i} + b) \ge 1 - \xi_{i}
# ```
```

File: docs/src/examples/sensitivity-analysis-svm.jl [hunk 4]
```diff
@@ -160,9 +134,10 @@ for Xi in 1:N
         DiffOpt.ForwardOutVariablePrimal(),
         b,
     )
-    push!(∇, norm(dw) + norm(db))
 
-    dy[Xi] = 0.0  # reset the change made above
+    ∇[i] = norm(dw) + norm(db)
+
+    dy[i] = 0.0  # reset the change made above
```
Comment (matbesancon) [inline docs/src/examples/sensitivity-analysis-svm.jl:140]:
```suggestion
    dy[i] = 0.0
```

File: docs/src/examples/sensitivity-analysis-svm.jl [hunk 5]
```diff
@@ -160,9 +134,10 @@ for Xi in 1:N
         DiffOpt.ForwardOutVariablePrimal(),
         b,
     )
-    push!(∇, norm(dw) + norm(db))
 
-    dy[Xi] = 0.0  # reset the change made above
+    ∇[i] = norm(dw) + norm(db)
+
```
Comment (matbesancon) [inline docs/src/examples/sensitivity-analysis-svm.jl:139]:
```suggestion
```

File: docs/src/examples/sensitivity-analysis-svm.jl [hunk 6]
```diff
@@ -160,9 +134,10 @@ for Xi in 1:N
         DiffOpt.ForwardOutVariablePrimal(),
         b,
     )
-    push!(∇, norm(dw) + norm(db))
```
Comment (matbesancon) [inline docs/src/examples/sensitivity-analysis-svm.jl:137]:
```suggestion
```

File: docs/src/examples/sensitivity-analysis-svm.jl [hunk 7]
```diff
@@ -94,59 +98,29 @@ Plots.plot!(p, svm_x, svm_y, label = "loss = $(round(loss, digits=2))", width=3)
...
-# begin differentiating
-for Xi in 1:N
-    dy[Xi] = 1.0  # set
+# Begin differentiating the model.
+# analogous to varying theta in the expression:
+# ```math
+# (y_{i} + \theta) (w^T X_{i} + b) \ge 1 - \xi_{i}
+# ```
+for i in 1:N
+    dy[i] = 1.0 # set
```
Comment (matbesancon) [inline docs/src/examples/sensitivity-analysis-svm.jl:115]:
```suggestion
```

File: docs/src/examples/sensitivity-analysis-svm.jl [hunk 8]
```diff
@@ -183,20 +158,22 @@ Plots.plot!(p2, svm_x, svm_y, label = "loss = $(round(loss, digits=2))", width=3
...
 
-∇ = Float64[]
+∇ = zeros(N)
 dX = zeros(N, D);
 
 # begin differentiating
-for Xi in 1:N
-    dX[Xi, :] = ones(D)  # set
+for i in 1:N
+    dX[i, :] = ones(D)  # set
```
Comment (matbesancon) [inline docs/src/examples/sensitivity-analysis-svm.jl:164]:
```suggestion
```

File: docs/src/examples/sensitivity-analysis-svm.jl [hunk 9]
```diff
@@ -94,59 +98,28 @@ Plots.plot!(p, svm_x, svm_y, label = "loss = $(round(loss, digits=2))", width=3)
...
+    for j in 1:N
+        MOI.set(
+            model,
+            DiffOpt.ForwardInConstraint(),
+            cons[j],
+            dy[j] * (dot(X[j,:], index.(w)) + index(b)),
+        )
+    end
 
     DiffOpt.forward(model)
```
Comment (matbesancon) [inline docs/src/examples/sensitivity-analysis-svm.jl:125]:
```suggestion
    end
    DiffOpt.forward(model)
```

**Comment by joaquimg** [comment]
Thanks for the fixes @matbesancon , I pushed and went to bed XD.

> thanks @joaquimg, the second experiment out of which and which?

We currently have:

Experiment 1: Gradient of hyperplane wrt the data point labels

and

Experiment 2: Gradient of hyperplane wrt the data point coordinates


I would get rid of "Experiment 1: Gradient of hyperplane wrt the data point labels", because results are too similar.
I don't think this is a major issue though. So if someone like the idea of having both I'd be fine.
